test/dataset/test_unsupervised_dataset.py

```diff
@@ -21,3 +24,26 @@ def test_unsupervised_waveform_dataset(libri_cut_set):
     assert len(dataset) == 1
     audio = dataset[0]
     assert audio.shape == (1, 10 * 16000)
+
+
+def test_on_the_fly_feature_extraction_unsupervised_dataset(libri_cut_set):
+    ref_dataset = UnsupervisedDataset(libri_cut_set)
+    tested_dataset = OnTheFlyFeatureExtractionUnsupervisedDataset(
+        feature_extractor=Fbank(),
+        cuts=libri_cut_set
+    )
+    ref_feats = ref_dataset[0]
+    tested_feats = tested_dataset[0]
+    # Note: comparison to 1 decimal fails.
+    #       I'm assuming this is due to lilcom's compression.
+    #       Pytest outputs looks like the following:
+    # E       Mismatched elements: 4 / 23000 (0.0174%)
+    # E       Max absolute difference: 0.46469784
+    # E       Max relative difference: 0.6171043
```

Comment:
Mm, lilcom's compression should not create this much difference, it should be rounding exactly to the closest 2^-5 given the current default, so this shouldn't be a lilcom issue.  Can you look more into this?